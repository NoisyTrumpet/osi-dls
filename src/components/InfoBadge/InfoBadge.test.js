import React from 'react';
import { render, screen } from '@testing-library/react';
import {
  toBeInTheDocument,
  toHaveClass,
  toHaveStyle,
  toHaveAttribute,
} from '@testing-library/jest-dom/matchers';
import { InfoBadge } from './';

expect.extend({
  toBeInTheDocument,
  toHaveClass,
  toHaveStyle,
  toHaveAttribute,
});

describe('<InfoBadge />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<InfoBadge preset="success" info="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  // Correct Class
  test('has correct class and color', () => {
    const { rerender } = render(<InfoBadge preset="success" info="test" />);
    expect(screen.getByText('test').closest('div')).toHaveClass('info-badge');
    rerender(<InfoBadge preset="success" info="test" />);
    expect(screen.getByText('test').closest('div')).toHaveStyle({
      color: 'rgb(44, 134, 54);',
    });
  });
});
