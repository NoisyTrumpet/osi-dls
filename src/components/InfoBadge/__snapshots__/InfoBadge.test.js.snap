// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoBadge /> renders and matches snapshot 1`] = `
<div
  class="info-badge"
>
  <span
    aria-hidden="true"
    class="icon icon-check-default-bold icon-util-success icon-15 info-badge-icon"
    data-testid="icon"
  />
  <span
    class="typography typography-caption color-util-success"
  >
    test
  </span>
</div>
`;
